### PR TITLE
CSS code hints in 'style' attribute value context

### PR DESCRIPTION
**Pull request by swmitra**
This PR enables Brackets to provide CSS code hints property-value code hints while editing 'style' attribute value in 'html' mode. The implementation contains minor tweaks to the CSSUtils module to recognize this as 'css' context as CodeMirror doesn't understand the 'style' attribute value as 'css' context. 

Adding @ingorichter @petetnt @zaggino and @ficristo for review, but others can also join 👍

### CODE REVIEW

**Comment by ficristo** [comment]
Sorry but I don't have enough interest in this feature to review it.
Maybe @humphd is interested?

**Comment by swmitra** [comment]
@humphd Can you please have a look at this PR?

File: src/language/CSSUtils.js [hunk 1]
```diff
@@ -613,16 +620,34 @@ define(function (require, exports, module) {
             return createInfo();
         }
 
+        // Context from the current editor will have htmlState if we are in css mode
+        // and in attribute value state of a tag with attribute name style
+        if (ctx.token.state.htmlState && (!ctx.token.state.localMode || ctx.token.state.localMode.name !== "css")) {
+
+                // tagInfo is required to aquire the style attr value
```
Comment (ingorichter) [inline src/language/CSSUtils.js]: nit: indent level

**Comment by ingorichter** [comment]
@swmitra it would be beneficial to have some unit tests for the new functionality. This is complex and might break accidentally.

**Comment by swmitra** [comment]
Sure @ingorichter. I will add some functional tests in this PR and ping you.

File: src/language/CSSUtils.js (hunk 1 above)
Comment (petetnt) [inline src/language/CSSUtils.js]: Indent

**Review by petetnt — CHANGES_REQUESTED** [review]
Change set LGTM with a small nit. 

However it doesn't seem to handle multiline styles yet:

```html
<div style="background-color: red; // works here
color: white; // doesn't work here>Foo</div>
```

It would be also great if it worked on JSX context (`<div style={{backgroundColor: red}}`>Hey</div>`) but that's more a feature request :)

**Comment by swmitra** [comment]
@ingorichter I have added some unit tests in my last commit.
@petetnt Fixed the multiline attribute value issue and corrected indentation. 
Can you guys please review it once more? ☺️

**Review by zaggino — APPROVED** [review]
- checked the code, seems good
- checked the unit tests, present & passing
- tested the feature manually, working fine

**Comment by ingorichter** [comment]
I'm too late :-/

**Pull request merged**